Code/GraphMol/CIPLabeler/CIPLabeler.cpp

```diff
@@ -0,0 +1,193 @@
+//
+//
+//  Copyright (C) 2020 Schrödinger, LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <algorithm>
+#include <memory>
+
+#include <GraphMol/RDKitBase.h>
+
+#include "CIPLabeler.h"
+#include "CIPMol.h"
+#include "configs/Sp2Bond.h"
+#include "configs/Tetrahedral.h"
+
+#include "rules/Rules.hpp"
+#include "rules/Rule1a.h"
+#include "rules/Rule1b.h"
+#include "rules/Rule2.h"
+#include "rules/Rule3.h"
+#include "rules/Rule4a.h"
+#include "rules/Rule4b.h"
+#include "rules/Rule4c.h"
+#include "rules/Rule5New.h"
+#include "rules/Rule6.h"
+
+namespace RDKit {
+namespace CIPLabeler {
+
+namespace {
+
+// constitutional rules
+const Rules constitutional_rules({new Rule1a, new Rule1b, new Rule2});
+
+// all rules (require aux calc)
+const Rules all_rules({new Rule1a, new Rule1b, new Rule2, new Rule3, new Rule4a,
+                       new Rule4b, new Rule4c, new Rule5New, new Rule6});
+
+std::vector<std::unique_ptr<Configuration>> findConfigs(const CIPMol &mol) {
+  std::vector<std::unique_ptr<Configuration>> configs;
+
+  for (auto &atom : mol.atoms()) {
+    auto chiraltag = atom->getChiralTag();
+    if (chiraltag == Atom::CHI_TETRAHEDRAL_CW ||
+        chiraltag == Atom::CHI_TETRAHEDRAL_CCW) {
+      std::vector<Atom *> carriers;
+      carriers.reserve(4);
+      for (auto &bond : mol.getBonds(atom)) {
+        auto nbr = bond->getOtherAtom(atom);
+        carriers.push_back(nbr);
+      }
+      if (carriers.size() < 4) {
+        // Implicit H -- use the central atom instead of a dummy H
+        carriers.push_back(atom);
+      }
+
+      auto atom_cfg = -1;
+      if (chiraltag == Atom::CHI_TETRAHEDRAL_CW) {
+        atom_cfg = Tetrahedral::RIGHT;
+      } else {
+        atom_cfg = Tetrahedral::LEFT;
+      }
+
+      std::unique_ptr<Tetrahedral> cfg{
+          new Tetrahedral(mol, atom, std::move(carriers), atom_cfg)};
+      configs.push_back(std::move(cfg));
+    }
+  }
+
+  for (auto i = 0u; i < mol.getNumBonds(); ++i) {
+    auto bond = mol.getBond(i);
+
+    int bond_cfg = -1;
+    switch (bond->getStereo()) {
+    case Bond::STEREOE:
+    case Bond::STEREOTRANS:
+      bond_cfg = Sp2Bond::OPPOSITE;
+      break;
+    case Bond::STEREOZ:
+    case Bond::STEREOCIS:
+      bond_cfg = Sp2Bond::TOGETHER;
+      break;
+    default:
+      continue;
+    }
+
+    auto stereo_atoms = MolOps::findStereoAtoms(bond);
```

Comment:
You are relying on the size of this later, so you should check here that the result has 2 elements.

Reply:
Yes, that makes sense. I'm adding an invariant check here.